examples/database_components/src/state.rs

```diff
@@ -10,7 +10,7 @@ use std::sync::Arc;
 
 #[auto_impl(&mut, Box)]
 pub trait State {
-    type Error;
+    type Error: core::error::Error + 'static;
```

Comment:
Why `'static`?

Reply:
in this case, it's required by the `impl :Error for DatabaseComponentError<T, U>` [here](https://github.com/bluealloy/revm/pull/1840/files#diff-f3bbb2a6ce7d66ff5946b45cd64d22824d61ce4c6708b9f55278b5e186426924R41). The bound could be removed here, but it would require all downstream users to propagate 2 bounds every time they want to use that `impl Error for DatabaseComponentError` (which is pretty often in practice)

Given that we don't have any examples of errors borrowing (i.e. all our current databases would meet this bound) it seems pretty reasonble to put it here too. This is just an example lib, and I don't feel strongly about its ergonomics tho

I would consider adding a `+ 'static` bound to the `Error` assoc type in `trait Database` tho, for the same reason. We have no clear use case for errors that capture data instead of own it, and it causes a lot of downstream complexity